portality/formcontext/fields.py

```diff
@@ -59,6 +59,9 @@ class URLField(TextField):
     widget = widgets.TextInput()
 
     def process_formdata(self, valuelist):
+        if valuelist is None or len(valuelist) == 0:
+             self.data = self.data = u''
```

Comment:
if you don't return early this will still fail on the next line when you try to access `valuelist[0]`

Reply:
Stupid mistake. I put that in if-else statement.